### PR TITLE
BUG-97678 Add V1 Connectors Networks test cases

### PR DESCRIPTION
**Pull request by mhalmy**
Please review it.

### CODE REVIEW

**Comment by mhmxs** [comment]
Could you please import config/idea/inspections.xml and fix inspection issues in the code?

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksProviderSpecificTests.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+                        }
+                    }
+                })
+        );
+    }
+
+    @AfterTest
+    public void cleanUp() throws Exception {
+        String[] nameArray = {VALID_CRED_NAME + "-az-spec", VALID_CRED_NAME + "-aws-spec", VALID_CRED_NAME + "-gcp-spec", VALID_CRED_NAME + "-os-spec"};
+        String errorMessage;
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksProviderSpecificTests.java]: Error message not used outside of catch

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                            credentialId = response.getId();
+                        }
+                    }
+                })
+        );
+    }
+
+    @AfterTest
+    public void cleanUp() throws Exception {
+        String errorMessage;
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java]: Error message not used outside of catch

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksProviderSpecificTests.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+    private final OpenstackCloudProvider openstackCloudProvider = new OpenstackCloudProvider(getTestParameter());
+
+    private final AwsCloudProvider awsCloudProvider = new AwsCloudProvider(getTestParameter());
+
+    private final AzureCloudProvider azureCloudProvider = new AzureCloudProvider(getTestParameter());
+
+    private final GcpCloudProvider gcpCloudProvider = new GcpCloudProvider(getTestParameter());
+
+    private String credentialName = "";
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksProviderSpecificTests.java]: This should be a local variable

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+    private static final String VALID_CRED_NAME = "valid-cred-";
+
+    private static final String INVALID_CRED_NAME = "invalid-cred";
+
+    private static final String INVALID_AV_ZONE = "invalid-avzone";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworksTest.class);
+
+    private String credentialName = "";
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java]: This should be a local variable

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/Networks.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    public static Assertion<Networks> assertThis(BiConsumer<Networks, IntegrationTestContext> check) {
+        return new Assertion<>(getTestContext(GherkinTest.RESULT), check);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Assertion<Networks> assertNameNotEmpty() {
+        return assertThis((networks, t) -> {
+            Set<Map.Entry<String, Set<PlatformNetworkResponse>>> networksEntrySet = networks.getNetworksResponse().getNetworks().entrySet();
+            for (Map.Entry entry : networksEntrySet) {
```
Comment (mhmxs) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/Networks.java]: Raw usage of generics type

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+    @BeforeTest()
+    public void createCredentialsAllProviders() throws  Exception {
+        credentialName = VALID_CRED_NAME + cloudProvider.getPlatform().toLowerCase() + "-nets";
+
+        given(CloudbreakClient.isCreated());
+        given(cloudProvider.aValidCredential()
+                .withName(credentialName)
+                .withDescription("")
+                .withCloudPlatform(cloudProvider.getPlatform()), credentialName + " credential is created.");
+        when(Credential.get(), credentialName + " credential has been got.");
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java]: We can spare the when/than steps in a BeforeTest method. I mean we have sufficient tests for credential, and the credential here should be "given"; the when then steps can focus on the network related things.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+                (credential, t) -> Assert.assertEquals(credential.getResponse().getName(), credentialName)),
+                credentialName + " should be the name of the new credential."
+        );
+        getCredentialId();
+    }
+
+    @Test()
+    @Parameters({ "region" })
+    public void testGetNetworksWithCredName(@Optional("") String region) throws Exception {
+        given(CloudbreakClient.isCreated());
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java]: For all tests beyond:
CloudbreakClient.isCreated is in the beforeTest, so already created ; there is no need here to duplicate

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+        when(Credential.get(), credentialName + " credential has been got.");
+        then(Credential.assertThis(
+                (credential, t) -> Assert.assertEquals(credential.getResponse().getName(), credentialName)),
+                credentialName + " should be the name of the new credential."
+        );
+        getCredentialId();
+    }
+
+    @Test()
+    @Parameters({ "region" })
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java]: you can use Cloudprovider.region() in test, for all provider it has a default, and you can parametrize (awsRegion, openstackRegion, etc)

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+    public void testGetNetworksWithCredName(@Optional("") String region) throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(Networks.request(createPlatformResourceRequest(null, credentialName, "", region)),
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java]: For four arguments a builder like .withXxx(x) method chaining might be more readable

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksProviderSpecificTests.java [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+        given(CloudbreakClient.isCreated());
+        given(Networks.request(NetworksTest.createPlatformResourceRequest(credentialIdMap.get("AZURE"), "", null, "")),
+                "get networks with invalid availability zone"
+        );
+        when(Networks.post(), "post the request");
+        then(Networks.assertNameNotEmpty());
+    }
+
+    @Test()
+    @Parameters({ "regionAWS" })
```
Comment (afarsang) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksProviderSpecificTests.java]: You can use AwsCloudprovider.region() if you will

**Pull request merged**